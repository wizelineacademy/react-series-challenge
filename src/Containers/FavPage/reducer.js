import {
  ADD_FAV,
  WITHDRAW_FAV
} from './action';

const initialState = {
  favGifs: [],
  favLoading: true
};

function favReducer(state = initialState, action) {
  
  switch (action.type) {
    case ADD_FAV:
      return {...state, favGifs: [...state.favGifs, action.params]}
    case WITHDRAW_FAV:
    const favGifs = state.favGifs.filter(fav => fav.id !== action.id);
      return {...state,favGifs}
    default:
      return state;
  }
}

export default favReducer;