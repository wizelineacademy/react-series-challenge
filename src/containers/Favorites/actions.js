import { createAction } from 'redux-actions';

const FAVORITE_ID_ADD = 'FAVORITE_ID_ADD';
const FAVORITE_ID_REMOVE = 'FAVORITE_ID_REMOVE';
const FAVORITES_LOAD = 'FAVORITES_LOAD';
const FAVORITES_LOADING_SET = 'FAVORITES_LOADING_SET';
const FAVORITES_SET = 'FAVORITES_SET';
const FAVORITES_SEARCH = 'FAVORITES_SEARCH';

const addFavoriteId = createAction(FAVORITE_ID_ADD);
const removeFavoriteId = createAction(FAVORITE_ID_REMOVE);
const loadFavorites = createAction(FAVORITES_LOAD);
const setLoadingFavorites = createAction(FAVORITES_LOADING_SET);
const setFavorites = createAction(FAVORITES_SET);
const searchFavorites = createAction(FAVORITES_SEARCH);

export default {
  types: {
    FAVORITE_ID_ADD,
    FAVORITE_ID_REMOVE,
    FAVORITES_LOAD,
    FAVORITES_LOADING_SET,
    FAVORITES_SET,
    FAVORITES_SEARCH,
  },
  creators: {
    addFavoriteId,
    removeFavoriteId,
    loadFavorites,
    setLoadingFavorites,
    setFavorites,
    searchFavorites,
  },
};
