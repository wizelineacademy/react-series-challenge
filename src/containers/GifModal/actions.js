import { createAction } from 'redux-actions';

const MODAL_OPEN = 'MODAL_OPEN';
const MODAL_CLOSE = 'MODAL_CLOSE';
const MODAL_GIF_SET = 'MODAL_GIF_SET';

const openModal = createAction(MODAL_OPEN);
const closeModal = createAction(MODAL_CLOSE);
const setModalGif = createAction(MODAL_GIF_SET);

export default {
  types: {
    MODAL_OPEN,
    MODAL_CLOSE,
    MODAL_GIF_SET,
  },
  creators: {
    openModal,
    closeModal,
    setModalGif,
  },
};
