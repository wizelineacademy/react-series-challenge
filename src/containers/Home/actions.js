import { createAction } from 'redux-actions';

const TRENDING_GIFS_LOAD = 'TRENDING_GIFS_LOAD';
const LOADING_SET = 'LOADING_SET';
const GIFS_SET = 'GIFS_SET';
const GIFS_SEARCH = 'GIFS_SEARCH';
const SEARCH_TERM_SET = 'SEARCH_TERM_SET';

const loadTrendingGifs = createAction(TRENDING_GIFS_LOAD);
const setLoading = createAction(LOADING_SET);
const setGifs = createAction(GIFS_SET);
const searchGifs = createAction(GIFS_SEARCH);
const setSearchTerm = createAction(SEARCH_TERM_SET);

export default {
  types: {
    TRENDING_GIFS_LOAD,
    LOADING_SET,
    GIFS_SET,
    GIFS_SEARCH,
    SEARCH_TERM_SET,
  },
  creators: {
    loadTrendingGifs,
    setLoading,
    setGifs,
    searchGifs,
    setSearchTerm,
  },
};
