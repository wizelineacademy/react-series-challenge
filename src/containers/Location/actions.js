import { createAction } from 'redux-actions';

const PATH_SET = 'PATH_SET';
const setPath = createAction(PATH_SET);

export default {
  types: {
    PATH_SET,
  },
  creators: {
    setPath,
  },
};
