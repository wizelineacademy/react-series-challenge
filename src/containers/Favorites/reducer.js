import actions from './actions';
const {
  FAVORITE_ID_ADD,
  FAVORITE_ID_REMOVE,
  FAVORITES_SET,
  FAVORITES_SEARCH,
} = actions.types;

const initialState = {
  favoritesIds: [],
  gifs: [],
  filteredGifs: [],
};

const reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case FAVORITE_ID_ADD:
      return { ...state, favoritesIds: [...state.favoritesIds, payload.id] };
    case FAVORITE_ID_REMOVE:
      const favoritesIds = state.favoritesIds.filter(id => {
        return id !== payload.id;
      });
      const gifs = state.gifs.filter(gif => {
        return gif.id !== payload.id;
      });
      const filteredGifs = state.filteredGifs.filter(gif => {
        return gif.id !== payload.id;
      });

      return {
        ...state,
        favoritesIds: favoritesIds,
        gifs: gifs,
        filteredGifs: filteredGifs,
      };
    case FAVORITES_SET:
      return { ...state, gifs: payload, filteredGifs: payload };
    case FAVORITES_SEARCH:
      let filtered = state.gifs;
      if (payload.term) {
        filtered = state.gifs.filter(gif => {
          return gif.title.indexOf(payload.term) > -1;
        });
      }
      return { ...state, filteredGifs: filtered };
    default:
      return state;
  }
};

export default reducer;
