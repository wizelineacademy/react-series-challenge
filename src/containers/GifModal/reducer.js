import actions from './actions';

const { MODAL_OPEN, MODAL_CLOSE, MODAL_GIF_SET } = actions.types;

const initialState = {
  isOpen: false,
  gif: null,
};

const reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case MODAL_OPEN:
      return { ...state, isOpen: true };
    case MODAL_CLOSE:
      return { ...state, isOpen: false };
    case MODAL_GIF_SET:
      return { ...state, gif: payload.gif };
    default:
      return state;
  }
};

export default reducer;
