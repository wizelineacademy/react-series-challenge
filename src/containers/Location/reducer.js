import actions from './actions';

const { PATH_SET } = actions.types;
const initialState = {
  path: '',
};
const reducer = (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {
    case PATH_SET:
      return { ...state, path: payload };
    default:
      return state;
  }
};

export default reducer;
