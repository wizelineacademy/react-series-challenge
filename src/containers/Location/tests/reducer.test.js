import reducer from '../reducer';
import actions from '../actions';

describe('Location container reducer', () => {
  it('should modifiy the reducer state path property when sending a PATH_SET action', () => {
    const action = actions.creators.setPath('/favorites');
    const newState = reducer(undefined, action);
    expect(newState).toMatchSnapshot();
  });

  it('should return the initialState or prevState when sending a not recognized action', () => {
    const action = { type: 'UNKNOWN.action' };
    const newState = reducer(undefined, action);
    expect(newState).toMatchSnapshot();
  });
});
