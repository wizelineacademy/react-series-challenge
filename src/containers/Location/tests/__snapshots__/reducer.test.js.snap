// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location container reducer should modifiy the reducer state path property when sending a PATH_SET action 1`] = `
Object {
  "path": "/favorites",
}
`;

exports[`Location container reducer should return the initialState or prevState when sending a not recognized action 1`] = `
Object {
  "path": "",
}
`;
